core/station/impl/src/factories/blockchains/internet_computer.rs

```diff
@@ -137,17 +156,59 @@ impl InternetComputer {
         Ok(balance.e8s())
     }
 
+    /// Returns the latest balance of the given icrc1 account of a station account.
+    pub async fn balance_of_icrc1_account(
+        &self,
+        asset: &Asset,
+        account: &icrc_ledger_types::icrc1::account::Account,
+    ) -> BlockchainApiResult<BigUint> {
+        let ledger_canister_id = Self::get_ledger_canister_id_from_metadata(&asset.metadata)?;
+
+        let balance =
+            ic_cdk::call::<(icrc_ledger_types::icrc1::account::Account,), (candid::Nat,)>(
+                ledger_canister_id,
+                "icrc1_balance_of",
+                // 4. Provide the arguments for the call in a tuple, here `transfer_args` is encapsulated as a single-element tuple.
+                (*account,),
+            )
+            .await
+            .map_err(|err| BlockchainApiError::BlockchainNetworkError {
+                info: format!("rejection_code: {:?}, err: {}", err.0, err.1),
+            })?
+            .0;
+
+        Ok(balance.0)
+    }
+
     pub fn transaction_fee(&self) -> u64 {
-        DEFAULT_FEE.e8s()
+        ic_ledger_types::DEFAULT_FEE.e8s()
```

Comment:
hardcoding this it assumes that we will only support ICP to be under this internet_computer implementation right? e.g. if OGY which also supports ICRC1 and ICP standard would want to use this they would not be able to if their fee would diverge

Reply:
good catch, will update this so it also fetches from the canister